src/components/UriViewer.js

```diff
@@ -52,7 +68,38 @@ const getMaxDownloadCostNA = bytes => {
   return downloadPrice < 0.01 ? '< $0.01' : Utils.formatUSD(downloadPrice)
 }
 
-const UriViewer = ajaxCaller(class UriViewer extends Component {
+const UriViewer = props => {
+  const signal = useCancellation()
+  const [valid, setValid] = useState(undefined)
+
+  const { googleProject, uri, onDismiss } = props
+  const isGsUri = isGs(uri)
+  const [bucket, name] = isGsUri ? parseUri(uri) : []
+
+  const pingObject = withErrorReporting('Error fetching file metadata', async () => {
+    if (isGsUri) {
+      try {
+        await Ajax(signal).Buckets.getObject(bucket, name, googleProject)
+        setValid(true)
+      } catch (e) {
+        const isRequesterPays = e.status === 400 && (await e.json()).error.message.includes('requester pays')
+        setValid(!isRequesterPays)
+      }
+    }
+  })
+
+  Utils.useOnMount(() => {
+    pingObject()
+  })
+
+  return Utils.switchCase(valid,
+    [undefined, () => h(Modal, viewerProps(onDismiss), loadingMetadata(uri))],
```

Comment:
Nit: I think I'd rather this be `{ ...viewerProps, onDismiss }` instead of making `viewerProps` a function. Same in the `UriViewerModal.render()` function below.